easybuild/tools/module_generator.py

```diff
@@ -213,18 +213,10 @@ def _filter_paths(self, key, paths):
             return paths
 
         added_paths = self.added_paths_per_key.setdefault(key, set())
-        # paths can be a string
-        if isinstance(paths, str):
-            if paths in added_paths:
-                filtered_paths = None
-            else:
-                added_paths.add(paths)
-                filtered_paths = paths
-        else:
-            # Coerce any iterable/generator into a list
-            if not isinstance(paths, list):
-                paths = list(paths)
-            filtered_paths = [x for x in paths if x not in added_paths and not added_paths.add(x)]
+        # Coerce any iterable/generator into a list
```

Comment:
I'm a bit worried about the change here. Why changing `_filter_paths` and even removing the ability to pass a string (which is a breaking change leading to surprising behavior)?

Reply:
The rest of the code (_update_paths) both converted the str case to a list. I simply moved this earlier, which also removes the need for _filter_paths to keep duplicate logic, since regardless it was always converted to a list immediately afterwards anyway.

You can pass a str to `paths`  to `append/prepend/update_paths`.

Reply:
Yes I see that **our** usages in **this** file basically make this part dead code. However someone or something else might already use this code, so I'm not sure if we should be a bit more careful here. E.g. at least error out when a string is passed instead of an iterable of strings

On the other hand: Calling this method from outside would have been wrong (it is "private") and this PR is targetting 5.x so a breaking change for simpler logic is fine.